soh/src/overlays/actors/ovl_En_Gb/z_en_gb.c

```diff
@@ -337,14 +337,18 @@ void func_80A2FA50(EnGb* this, PlayState* play) {
         Player_UpdateBottleHeld(play, GET_PLAYER(play), ITEM_BOTTLE, PLAYER_IA_BOTTLE);
         Rupees_ChangeBy(50);
         HIGH_SCORE(HS_POE_POINTS) += 100;
-        if (HIGH_SCORE(HS_POE_POINTS) != 1000) {
+        if (
+            (!gSaveContext.n64ddFlag && HIGH_SCORE(HS_POE_POINTS) != 1000) ||
+            (gSaveContext.n64ddFlag && (HIGH_SCORE(HS_POE_POINTS) != 1000 || Flags_GetRandomizerInf(RAND_INF_10_BIG_POES)))
```

Comment:
In the fishing check you are strictly checking only for the RAND_INF flag being set or not while ignoring the HIGH_SCORE, but here you are OR-ing them. Can you explain why one is one way and the other is different?

Reply:
@Archez The result of this condition being truthy is the fail case (as in, don't give item)

So in rando, they will only receive the item if they haven't received it yet AND the points is exactly 1000